packages/block-editor/src/hooks/border-width.js

```diff
@@ -1,16 +1,17 @@
 /**
  * WordPress dependencies
  */
-import { RangeControl } from '@wordpress/components';
+import { __experimentalUnitControl as UnitControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import { CSS_UNITS, parseUnit } from './border';
 import { cleanEmptyObject } from './utils';
 
 const MIN_BORDER_WIDTH = 0;
-const MAX_BORDER_WIDTH = 50;
```

Comment:
Are we getting rid of max for any reason or because it doesn't play well with varying units?

Reply:
There were two main reasons behind me making that change. First, was the max was required to utilise the slider control in the original border radius support UI. Second, a max radius of `50px` wasn't very flexible at all and at worst simply didn't make sense in some contexts.

Given a max value wasn't required for the `UnitControl` and that the max would need to change based on context e.g. type of block, dimensions of the block etc, I saw only minor downside in removing the max. The rounded block style for the image block for example uses `border-radius: 9999px`.

I'm happy to iterate on this in future if we do want to constrain values more. We'd need some means in the block support mechanism to be able to provide the correct max for the context. It could be a flag within the block.json support configuration for example.